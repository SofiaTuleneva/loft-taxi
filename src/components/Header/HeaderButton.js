import React from 'react'
import PropTypes from 'prop-types'

const HeaderButton = ({handleClick, text}) => {
	return (
		<button onClick={handleClick} className="menu__item">
			{text}
		</button>
	)
};

HeaderButton.propTypes = {
	handleClick: PropTypes.func.isRequired,
	text: PropTypes.string.isRequired,
};

export default HeaderButton;