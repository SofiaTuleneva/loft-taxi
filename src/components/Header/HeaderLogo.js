import React from 'react'
import HeaderIcon from '../../img/logo.svg'
import {Typography} from "@material-ui/core";


const HeaderLogo = () => {
    return (
        <Typography>
            <img src={HeaderIcon} width="156" alt="logo" className="header__pic"/>
        </Typography>
    )
};

export default HeaderLogo;