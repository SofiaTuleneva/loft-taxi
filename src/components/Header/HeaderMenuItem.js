import React from 'react'
import PropTypes from 'prop-types'

const HeaderMenuItem = ({handleClick, text}) => {
	return (
		<button onClick={handleClick} className="menu__item">
			{text}
		</button>
	)
};

HeaderMenuItem.propTypes = {
	handleClick: PropTypes.func.isRequired,
	text: PropTypes.string.isRequired,
};

export default HeaderMenuItem;