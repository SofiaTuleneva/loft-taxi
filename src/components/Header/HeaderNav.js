import React from 'react';
import {logoutAction} from "../../modules/auth/actions";
import {useDispatch} from "react-redux";
import {paths} from '../../constants/Paths';
import {Button} from "@material-ui/core";

const HeaderNav = () => {
	const dispatch = useDispatch();

	const handleLogout = e => {
		e.preventDefault();
		dispatch(logoutAction());
	};

	return (
		<div>
			<Button href={paths.map}>Карта</Button>
			<Button href={paths.profile}>Профиль</Button>
			<Button onClick={handleLogout}>Выйти</Button>
		</div>
	);
};

export default HeaderNav;