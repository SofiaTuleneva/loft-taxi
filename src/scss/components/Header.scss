.header {
  .header__appbar {
    z-index: 10000;
    width: 100%;
    background-color: $white;
    position: relative;
    animation: jss1 0.5s cubic-bezier(0.230, 1.000, 0.320, 1.000) both 0.3s;
    box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12)
  }

  .header__toolbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
}