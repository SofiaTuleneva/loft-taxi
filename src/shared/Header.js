import React from 'react';

class Header extends React.Component {

	handleMenuClick = (e, pageId) => {
		e.preventDefault();
		this.props.changePage(pageId)
	};

	render() {
		return (
			<div className="Header">
				{this.props.pages.map(({ id, name }) => (
					<button
						key={id}
						onClick={e => this.handleMenuClick(e, id)}
						className={id === this.props.activePageId ? 'activeButton' : null}
					>
					{name}
					</button>
				))}
			</div>
		);
	}

}

export default Header;
