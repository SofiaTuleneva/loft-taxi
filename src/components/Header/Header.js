import React, {useContext} from 'react';
import PropTypes from 'prop-types'
import UserContext from "../../context/UserContext";
import HeaderLogo from "./HeaderLogo";
import HeaderButton from "./HeaderButton";

const Header = ({pages, setPage}) => {

	// Context
	const user = useContext(UserContext);

	const handleMenuClick = (e, activePageId) => {
		e.preventDefault();
		setPage(activePageId)
	};

	const handleLogout = e => {
		e.preventDefault();
		user.logout();
	};

	const pagesFiltered = pages.filter(
		({showInMenu}) => showInMenu === user.isLoggedIn
	);

	return (
		<header className="header">
			<div className="container">
				<div className="header__line">
					<HeaderLogo/>
					<div className="header__menu">
						{pagesFiltered.map(({id, text}) => (
							<HeaderButton
								key={id}
								text={text}
								handleClick={e => handleMenuClick(e, id)}
							/>
						))}
						<button onClick={handleLogout} className="menu__item">Выйти</button>
					</div>
				</div>
			</div>
		</header>
	);
};

Header.propTypes = {
	pages: PropTypes.array.isRequired,
	setPage: PropTypes.func.isRequired,
};

export default Header;
