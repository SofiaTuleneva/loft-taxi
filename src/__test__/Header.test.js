import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
	Map: () => ({})
}));

it("Header renders correctly", () => {
	const setPage = jest.fn();
	const component = renderer.create(
		<Header
			setPage={setPage}
			activePageId={'login'}
			pages={[]}
		/>,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
