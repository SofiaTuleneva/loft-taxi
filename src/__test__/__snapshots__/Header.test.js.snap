// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__line"
    >
      <div
        className="header__logo"
      >
        <img
          alt="logo"
          className="header__pic"
          src="logo.svg"
          width="156"
        />
      </div>
      <div
        className="header__menu"
      >
        <button
          className="menu__item"
          onClick={[Function]}
        >
          Выйти
        </button>
      </div>
    </div>
  </div>
</header>
`;
